### PR TITLE
Restructuring to improve prove properties

### PR DESCRIPTION
**Pull request by jklmnn**
- move session property functions to parent package
 - add initialized property to client instance
 - enable and check the usage of preconditions on generic formal argument functions
 - check that requests are only used by the session that created them

### CODE REVIEW

File: src/block/server/componolit-interfaces-block-dispatcher.ads [hunk 1]
```diff
@@ -120,4 +109,22 @@ is
       Pre  => Initialized (D),
       Post => Initialized (D);
 
+private
+
+   --  Enforces the precondition of Dispatch
+   --
+   --  The only valid precondition for Dispatch is Initialized (D). This enforced by this
```
Comment (senier) [inline src/block/server/componolit-interfaces-block-dispatcher.ads]: This **is** enforced

File: test/block_proxy/component.adb [hunk 2]
```diff
@@ -40,33 +36,42 @@ package body Component is
...
+                                   S => Block.Null_Request,
                                    A => False));
 
    procedure Write (C :     Block.Client_Instance;
                     I :     Request_Index;
                     D : out Buffer)
    is
-      pragma Unreferenced (C);
+      use type Block.Client_Instance;
+      --  pragma Unreferenced (C);
```
Comment (senier) [inline test/block_proxy/component.adb]: Remove?

File: test/block_proxy/component.adb [hunk 3]
```diff
@@ -75,77 +80,108 @@ package body Component is
...
                         Cache (I).A := True;
                   end case;
                end if;
-               if Block_Client.Status (Cache (I).C) = Block.Allocated then
+               if Block.Status (Cache (I).C) = Block.Allocated then
                   Block_Client.Enqueue (Client, Cache (I).C);
+                  null;
                end if;
             end if;
+            --  pragma Assert (Block.Initialized (Server));
```
Comment (senier) [inline test/block_proxy/component.adb]: ???

Comment (jklmnn) [reply test/block_proxy/component.adb]: Fixed by https://github.com/Componolit/ada-interfaces/pull/83/commits/3395b9ae71b786b4d29e9e85c56afb7e5b15568f.

File: test/block_proxy/component.adb (hunk 2 above)
Comment (jklmnn) [reply test/block_proxy/component.adb]: Fixed by https://github.com/Componolit/ada-interfaces/pull/83/commits/3395b9ae71b786b4d29e9e85c56afb7e5b15568f.

File: src/block/server/componolit-interfaces-block-dispatcher.ads (hunk 1 above)
Comment (jklmnn) [reply src/block/server/componolit-interfaces-block-dispatcher.ads]: Fixed by https://github.com/Componolit/ada-interfaces/pull/83/commits/3395b9ae71b786b4d29e9e85c56afb7e5b15568f.

**Pull request merged**